core/src/main/java/org/infinispan/compat/BiFunctionMapper.java

```diff
@@ -0,0 +1,59 @@
+package org.infinispan.compat;
+
+import java.io.IOException;
+import java.io.ObjectInput;
+import java.io.ObjectOutput;
+import java.util.Collections;
+import java.util.Set;
+import java.util.function.BiFunction;
+
+import org.infinispan.commons.marshall.AdvancedExternalizer;
+import org.infinispan.commons.marshall.Ids;
+import org.infinispan.factories.annotations.Inject;
+
+public class BiFunctionMapper<K, V> implements BiFunction<K, V, V> {
```

Comment:
Hmm, writing the other comment I've noticed that the generics aren't right here: it should be
```
public class BiFunctionMapper<K, V, KT, VT> implements BiFunction<KT, VT, VT> {
   private transient TypeConverter<K, V, KT, VT> converter;
   private final BiFunction<? super K, ? super V, ? extends V> biFunction;
   ...
}
```
and the same incorrect generics usage in `FunctionMapper`.